pallets/loans/src/loan.rs

```diff
@@ -411,39 +411,34 @@ impl<T: Config> Pallet<T> {
             .ok_or(Error::<T>::OracleCurrencyPriceNotReady)?;
 
         //the total value of borrower's collateral token
-        let collateral_value = collateral_underlying_amount
-            .checked_mul(collateral_token_price)
+        let collateral_value = collateral_token_price
+            .checked_mul_int(collateral_underlying_amount)
             .ok_or(Error::<T>::CollateralOverflow)?;
 
         //calculate liquidate_token_sum
         let (liquidate_token_price, _) = T::PriceFeeder::get_price(&liquidate_currency_id)
             .ok_or(Error::<T>::OracleCurrencyPriceNotReady)?;
 
-        let liquidate_value = repay_amount
-            .checked_mul(liquidate_token_price)
+        let liquidate_value = liquidate_token_price
+            .checked_mul_int(repay_amount)
             .ok_or(Error::<T>::LiquidateValueOverflow)?;
 
         // the incentive for liquidator and punishment for the borrower
         let liquidation_incentive = LiquidationIncentive::<T>::get(liquidate_currency_id);
-        let discd_collateral_value =
-            mul_then_div(collateral_value, liquidation_incentive, RATE_DECIMAL)
-                .ok_or(Error::<T>::CalcDiscdCollateralValueFailed)?;
+        let discd_collateral_value = liquidation_incentive * collateral_value;
 
         if discd_collateral_value < liquidate_value {
             return Err(Error::<T>::RepayValueGreaterThanCollateral.into());
         }
 
         // calculate the collateral will get
-        let equivalent_collateral_amount = liquidate_value
-            .checked_div(collateral_token_price)
+        let equivalent_collateral_amount = collateral_token_price
+            .reciprocal()
+            .and_then(|price| price.checked_mul_int(liquidate_value))
```

Comment:
`price` should be `reciprocal_of_price`, and you can use `r` for short.

Reply:
fixed